backend/comment-section.go

```diff
@@ -0,0 +1,169 @@
+type AuthorizationObject struct {
+	User string `json:"user"`
+}
+
+type CommentSection struct {
+	Comments []Comment
+}
+
+type Comment struct {
+	Text     string
+	Datetime string
+}
+
+type AccessData struct {
+	ReaderID  string
+	ReturnURL string
+}
+
+func InitCommentSection() {
+	http.HandleFunc(COMMENT_SAMPLE_PATH+"submit-comment-xhr", func(w http.ResponseWriter, r *http.Request) {
+		handlePost(w, r, submitCommentXHR)
+	})
+	http.HandleFunc(COMMENT_SAMPLE_PATH+"submit-comment", func(w http.ResponseWriter, r *http.Request) {
+		handlePost(w, r, submitComment)
+	})
+	registerCommentSectionHandler("comment_section")
+	registerCommentSectionHandler("comment_section/preview")
+	http.HandleFunc(COMMENT_SAMPLE_PATH+"authorization", handleCommentAuthorization)
+	http.HandleFunc(COMMENT_SAMPLE_PATH+"pingback", handleCommentPingback)
+	http.HandleFunc(COMMENT_SAMPLE_PATH+"login", handleCommentLogin)
+	http.HandleFunc(COMMENT_SAMPLE_PATH+"logout", handleCommentLogout)
+	http.HandleFunc(COMMENT_SAMPLE_PATH+"submit", handleCommentSubmit)
+}
+
+func registerCommentSectionHandler(sampleName string) {
+	filePath := path.Join(DIST_FOLDER, SAMPLE_TEMPLATE_FOLDER, sampleName, "index.html")
+	template, err := template.New("index.html").Delims("[[", "]]").ParseFiles(filePath)
+	if err != nil {
+		panic(err)
+	}
+	route := path.Join(SAMPLE_TEMPLATE_FOLDER, sampleName) + "/"
+	http.HandleFunc(route, func(w http.ResponseWriter, r *http.Request) {
+		renderComment(w, r, sampleName, *template)
+	})
+}
+
+func renderComment(w http.ResponseWriter, r *http.Request, sampleName string, t template.Template) {
+	commentSection := listComment(sampleName, r)
+	w.Header().Set("Cache-Control", fmt.Sprintf("max-age=%d, public, must-revalidate", MAX_AGE_IN_SECONDS))
+	t.Execute(w, commentSection)
+}
+
+func listComment(sampleName string, r *http.Request) CommentSection {
+	var comments []Comment
+	lastComment := Comment{"This is the second comment", time.Now().Format("15:04:05")}
+	firstComment := Comment{"This is the first comment", time.Now().Add(time.Duration(MINUS_TEN_SECONDS) * time.Second).Format("15:04:05")}
+	comments = append(comments, lastComment)
+	comments = append(comments, firstComment)
+	return CommentSection{Comments: comments}
+}
+
+func submitCommentXHR(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("AMP-Access-Control-Allow-Source-Origin", buildSourceOrigin(r.Host))
+	w.Header().Set("Content-Type", "application/json")
+	response := ""
+	text := r.FormValue("text")
+	if text != "" {
+		newComment := Comment{
+			Text:     text,
+			Datetime: time.Now().Format("15:04:05"),
+		}
+		response = fmt.Sprintf("{\"Datetime\":\"%s\", \"Text\":\"%s\"}", newComment.Datetime, newComment.Text)
+		w.Write([]byte(response))
+	} else {
+		w.WriteHeader(http.StatusBadRequest)
+	}
+}
+
+func submitComment(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusBadRequest)
+}
+
+func handleCommentAuthorization(w http.ResponseWriter, r *http.Request) {
```

Comment:
@rowan-m Have a look at the latest changes on the server side, especially hotel.go and amp-access.go